modules/local/bedtools_genomic_origin_of_reads.nf

```diff
@@ -0,0 +1,72 @@
+process BEDTOOLS_GENOMIC_ORIGIN_OF_READS {
+    tag "$meta.id"
+    label 'process_medium'
+
+    container "flomicsbiotech/flomics_qc_rnaseq:dev"
+
+    input:
+    tuple val(meta), path(bam)
+    path  gtf
+
+    output:
+    tuple val(meta), path("*.txt"), emit: results
+    tuple val(meta), path("*_genomic_origin_of_reads.tsv"), emit: table
+    tuple val(meta), path("*_mqc.yaml"), emit: mqc_bedtools_goor_yaml
+    path  "versions.yml"          , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    shell:
+    '''
+    #take only exon records
+    awk '$3 == "exon"' !{gtf} > filtered_annotation_exon.gtf
+
+    #extract gene locus coordinates
+    cat !{gtf} | extract_locus_coords.pl - > gencode.genes.bed
+
+    #mapped fragments
+    samtools view -F2304 !{bam} | cut -f1 | sort | uniq > !{meta.id}_mapped_fragments.list.txt
```

Comment:
I think it would be better practice to first create a temporary BAM using the appropriate -F option (output to /tmp/, for example), and then use this as input to other processes. This way, if we have to modify the -F value in the future, we only have to touch one line of code.
I suggest something like:
```
uuid=$(uuidgen)
samtools view -H input.bam > /tmp/$uuid.sam
samtools view -F2304 input.bam >> /tmp/$uuid.sam
samtools view -b /tmp/tmp.sam > /tmp/$uuid.-F2304.bam
```